src/services/devHandler/updateUserScoreOnDailyProcessPhase1.js

```diff
@@ -0,0 +1,29 @@
+const {successResponse, errorResponse} = require('../../utils');
+const {
+  updateUserScoreOnDailyProcessPhase1
+} = require('../../processes/scoring/update-user-score-on-daily-process-phase1');
+const {getDb} = require('../../config/mongodb_conn');
+const {DB_COLLECTION_USER_SCORE} = require('../../processes/scoring-constant');
+
+const updateUserScorePhase1 = async (req, res) => {
+  try {
+    const {process_time, user_ids, last_batch} = req.body;
+    const db = await getDb();
+    const userScoreCol = await db.collection(DB_COLLECTION_USER_SCORE);
+    const result = await updateUserScoreOnDailyProcessPhase1(
+      null,
+      userScoreCol,
+      process_time,
+      user_ids,
+      last_batch
+    );
+    return successResponse(res, 'updateUserScoreOnDailyProcessPhase1 sucesfully', result);
+  } catch (error) {
+    console.log(error);
+    return errorResponse(res, error.toString(), 500);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `updateUserScorePhase1` lacks input validation for the request body. It's important to validate incoming data to prevent potential issues such as type errors, SQL injection, etc. Here is a suggestion on how you could add some basic validation:

```diff
+ const Joi = require('joi');

  const updateUserScorePhase1 = async (req, res) => {
    try {
+     const schema = Joi.object({
+       process_time: Joi.date().required(),
+       user_ids: Joi.array().items(Joi.string()).required(),
+       last_batch: Joi.boolean().required()
+     });
+     const { error } = schema.validate(req.body);
+     if (error) {
+       throw new Error(error.details[0].message);
+     }

      const {process_time, user_ids, last_batch} = req.body;
      const db = await getDb();
      const userScoreCol = await db.collection(DB_COLLECTION_USER_SCORE);
      const result = await updateUserScoreOnDailyProcessPhase1(
        null,
        userScoreCol,
        process_time,
        user_ids,
        last_batch
      );
      return successResponse(res, 'updateUserScoreOnDailyProcessPhase1 sucesfully', result);
    } catch (error) {
      console.log(error);
      return errorResponse(res, error.toString(), 500);
    }
  };
```
This uses the `Joi` library to define a schema for the expected structure of `req.body`, and then validates `req.body` against this schema. If the validation fails, an error is thrown with a message describing the issue.



<!-- This is an auto-generated comment by OSS CodeRabbit -->